src/main/java/frc/lib/lib2706/ProfiledPIDFFController.java

```diff
@@ -0,0 +1,55 @@
+package frc.lib.lib2706;
+
+import edu.wpi.first.math.controller.ProfiledPIDController;
+import edu.wpi.first.math.controller.SimpleMotorFeedforward;
+import edu.wpi.first.math.trajectory.TrapezoidProfile;
+import edu.wpi.first.wpilibj.Timer;
+import frc.robot.Config;
+import frc.robot.Config.ArmConfig;
+
+public class ProfiledPIDFFController {
```

Comment:
The design of this file is not the best practice but we can deal with that some other time. Ideally, any extra supporting files made for a subsystem should be designed to be generic and reusable each year. This file shouldn't need to reference anything from Config or others. So maybe that means you pass the ProfiledPidController and SimpleMotorFeedforwards through the constructor. Ideally, though you can switch to using the [ProfiledExternalPidController](https://github.com/FRC2706/2023-2706-Fall-Poseidon/blob/Erik-ArmSim/src/main/java/frc/lib/lib2706/ProfiledExternalPIDController.java) entirely because that reduces a few unneeded parts from this file as well. ProfiledExternalPIDController also has a really nice atGoal method: `public boolean atGoal(double positionTolerance, double velocityTolerance)`

https://github.com/FRC2706/2023-2706-Fall-Poseidon/blob/Erik-ArmSim/src/main/java/frc/lib/lib2706/ProfiledExternalPIDController.java

Reply:
I will copy the ProfiledExternalPidController to the 2024 repo for future uses